xarray/core/weighted.py

```diff
@@ -54,6 +65,48 @@
         New {cls} object with the sum of the weights over the given dimension.
     """
 
+_WEIGHTED_QUANTILE_DOCSTRING_TEMPLATE = """
+    Apply a a weighted ``quantile`` to this {cls}'s data along some dimension(s).
```

Comment:
It would be useful to specify what _kind_ of weight this function supports.
There are several definition of statistical weights and, for quantiles depending on the definition used the result will differ.
This article [1] exposes three of them:
- frequency weights (a.k.a fweight in np)
- sampling weights (a.k.a aweight in np), this PR implements it.
- precision weights (not sure if it can be used with quantiles) 


If we look at numpy [2], the intentions is likely to implement frequency weights (fweight) first.
However there are several PR on the subject and some [3] are targeting sampling weights (aweight).

I think it’s fine to handle a single kind of weight but it should be clear what definition is used.

----

1] Blog post about different weights: https://notstatschat.rbind.io/2020/08/04/weights-in-statistics/
The author is Thomas Lumley, a statistical researcher.

[2] Numpy PR 1 for weighted quantile: https://github.com/numpy/numpy/pull/9211#issuecomment-983705221

[3] Numpy PR 2 for weighted quantile: https://github.com/numpy/numpy/pull/16862. 
There is a third implementation in C in progress, but I can't find it.

Reply:
See new docstring paragraph.

Reply:
Since  the details are important here I think we should add
1. https://aakinshin.net/posts/weighted-quantiles
2. https://notstatschat.rbind.io/2020/08/04/weights-in-statistics/
3. Hyndman and Fan (1996): https://doi.org/10.2307/2684934

in a [References section](https://numpydoc.readthedocs.io/en/latest/format.html#references)

We can also leave this to a future PR.

Reply:
Done

Reply:
Wonderful  thanks!